iconservice/iconscore/icon_score_base.py

```diff
@@ -405,30 +405,44 @@ def __call(self,
                arg_params: Optional[list] = None,
                kw_params: Optional[dict] = None) -> Any:
 
-        if func_name != STR_FALLBACK:
-            self.validate_external_method(func_name)
-
-        self.__check_payable(func_name, self.__get_attr_dict(CONST_CLASS_PAYABLES))
-
-        score_func = getattr(self, func_name)
-
         if func_name == STR_FALLBACK:
+            if self._context.revision >= REVISION_4:
+                if not self.__is_payable_method(func_name):
+                    raise ExternalException(f"Method not found",
+                                            func_name,
+                                            type(self).__name__)
+            else:
+                if not self.__is_payable_method(func_name) and self.msg.value > 0:
+                    raise PayableException(f"This is not payable", func_name, type(self).__name__)
+
+            score_func = getattr(self, func_name)
             ret = score_func()
         else:
+            self.validate_external_method(func_name)
+            self.__check_payable(func_name, self.__get_attr_dict(CONST_CLASS_PAYABLES))
+            score_func = getattr(self, func_name)
             if arg_params is None:
                 arg_params = []
             if kw_params is None:
                 kw_params = {}
             ret = score_func(*arg_params, **kw_params)
-
         return ret
 
     def __check_payable(self, func_name: str, payable_dict: dict):
         if func_name not in payable_dict:
```

Comment:
I think it would be better to check `self.msg.value > 0` condition first.

Reply:
got it. fix check logic order